util/template/expression_template.go

```diff
@@ -0,0 +1,47 @@
+package template
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/Masterminds/sprig"
+	"github.com/antonmedv/expr"
+	"github.com/doublerebel/bellows"
+)
+
+const kindExpression kind = "="
+
+func init() {
+	if os.Getenv("EXPRESSION_TEMPLATES") != "false" {
+		kinds = append(kinds, kindExpression)
+	}
+}
+
+func expressionReplace(w io.Writer, expression string, env map[string]interface{}, allowUnresolved bool) (int, error) {
+	result, err := expr.Eval(expression, env)
+	if (err != nil || result == nil) && allowUnresolved { //  <nil> result is also un-resolved, and any error can be unresolved
+		return w.Write([]byte(fmt.Sprintf("{{%s%s}}", kindExpression, expression)))
+	}
+	if err != nil {
+		return 0, fmt.Errorf("failed to evaluate expression: %w", err)
+	}
+	if result == nil {
```

Comment:
Can `err` and `result` both be `nil` at the same time? If so, we may lose some important error information in one of them.

Reply:
yes - `result == nil` is the only way I could find to identify when the expression contains a variable that does not exist